networkx/generators/trees.py

```diff
@@ -184,11 +184,69 @@ def random_tree(n, seed=None):
     *n* nodes, the tree is chosen uniformly at random from the set of
     all trees on *n* nodes.
 
+    Example
+    -------
+    >>> import networkx as nx
+    >>> tree = nx.random_tree(n=10, seed=0)
+    >>> print(nx.forest_str(tree, sources=[0]))
+    ╙── 0
+        ├── 3
+        └── 4
+            ├── 6
+            │   ├── 1
+            │   ├── 2
+            │   └── 7
+            │       └── 8
+            │           └── 5
+            └── 9
+
+    >>> import networkx as nx
+    >>> tree = nx.random_tree(n=10, seed=0, create_using=nx.OrderedDiGraph)
+    >>> print(nx.forest_str(tree))
+    ╙── 0
+        ├─╼ 3
+        └─╼ 4
+            ├─╼ 6
+            │   ├─╼ 1
+            │   ├─╼ 2
+            │   └─╼ 7
+            │       └─╼ 8
+            │           └─╼ 5
+            └─╼ 9
     """
     if n == 0:
         raise nx.NetworkXPointlessConcept("the null graph is not a tree")
     # Cannot create a Prüfer sequence unless `n` is at least two.
     if n == 1:
-        return nx.empty_graph(1)
-    sequence = [seed.choice(range(n)) for i in range(n - 2)]
-    return nx.from_prufer_sequence(sequence)
+        utree = nx.empty_graph(1)
+    else:
+        sequence = [seed.choice(range(n)) for i in range(n - 2)]
+        utree = nx.from_prufer_sequence(sequence)
+
+    if create_using is None:
+        tree = utree
+    else:
+        # TODO: maybe a tree classmethod like
+        # Graph.new, Graph.fresh, or something like that
+        def new(cls_or_self):
+            if hasattr(cls_or_self, "_adj"):
+                # create_using is a NetworkX style Graph
+                cls_or_self.clear()
+                self = cls_or_self
+            else:
+                # try create_using as constructor
+                self = cls_or_self()
+            return self
+
+        tree = new(create_using)
```

Comment:
```suggestion
        tree = empty_graph(0, create_using)
```
As I understand it, the `new` function here is to handle the case that users can pass in classes *or* instances to `create_using`. This is hard to parse and I think it would be an improvement if specialized input-handling code could be avoided. Is the suggestion sufficient to accomplish the same behavior?

Side note: this is probably a good example of the motivation for [NXEP 3](https://networkx.org/documentation/latest/developer/nxeps/nxep-0003.html) - a code snippet like this might make a nice addition to the NXEP. CC @boothby

Reply:
I'll have to look into this more later. NXEP seems detailed, and I'd like to read it before I comment in detail.

I will say that the code in "new" exists in other places in networkx. I copied the pattern. I think I was going to propose putting the "new" logic into a classmethod. Perhaps that's addressed in the NXEP or in empty_graph. I'll take a look.

Reply:
I did accept @dschult's suggestion which is similar to this, but also includes removal of the above comment.